velox/exec/tests/utils/PrestoQueryRunner.h

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <folly/SocketAddress.h>
+#include <folly/io/async/EventBaseThread.h>
+#include "velox/common/memory/Memory.h"
+#include "velox/exec/tests/utils/ReferenceQueryRunner.h"
+#include "velox/vector/ComplexVector.h"
+
+namespace facebook::velox::exec::test {
+class PrestoQueryRunner : public velox::exec::test::ReferenceQueryRunner {
+ public:
+  /// @param coordinatorUri Presto REST API endpoint, e.g. http://127.0.0.1:8080
+  /// @param user Username to use in X-Presto-User header.
+  /// @param timeout Timeout in milliseconds of an HTTP request.
+  PrestoQueryRunner(
+      std::string coordinatorUri,
+      std::string user,
+      std::chrono::milliseconds timeout = std::chrono::milliseconds{1000});
+
+  /// Converts Velox query plan to Presto SQL. Supports Values -> Aggregation or
+  /// Window with an optional Project on top.
+  ///
+  /// Values node is converted into reading from 'tmp' table.
+  ///
+  /// @return std::nullopt if Values node uses types not supported by DWRF file
+  /// format (DATE, INTERVAL, UNKNOWN).
+  std::optional<std::string> toSql(
+      const velox::core::PlanNodePtr& plan) override;
+
+  /// Creates 'tmp' table using specified data, executes SQL query generated by
+  /// 'toSql' and returns the results.
+  ///
+  /// @param sql SQL generated by 'toSql' method.
+  /// @param input Data used in the Values node in the plan passed to 'toSql'
+  /// method.
+  /// @param resultType Expected type of the results.
+  /// @return Data received from Presto.
+  std::multiset<std::vector<velox::variant>> execute(
```

Comment:
nit: maybe in a future pr we can add in the option to pass in table name, so we can potentially have multiple queries running at same time. This can only run single threaded atm.

Reply:
OK, got it.

Reply:
@kgpai How about generating a random table name when constructing the `PrestoQueryRunner`, say, `tmp_$randonStringName_$threadId`?

Reply:
Let's keep things simple and first change the CI to allow running PrestoQueryRunnerTest, then replace Duck runner in the fuzzer.